src/Game/UI/Gumps/HealthBarGumpCustom.cs

```diff
@@ -0,0 +1,662 @@
+// the current name. You will need to type enter and re-pull the bar to see the name change.
+//
+// Lastly, I want to give thanks to KaRaShO, Roxya, Stalli, Gaechti, and Link for giving me tips on how
+// to get these Health Bars to function per my own vision; gratitude.
+
+#endregion
+
+using System;
+using System.IO;
+
+using ClassicUO.Game.Data;
+using ClassicUO.Game.GameObjects;
+using ClassicUO.Game.Managers;
+using ClassicUO.Game.UI.Controls;
+using ClassicUO.Input;
+using ClassicUO.IO;
+using ClassicUO.IO.Resources;
+using ClassicUO.Network;
+using ClassicUO.Renderer;
+
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using SDL2;
+
+namespace ClassicUO.Game.UI.Gumps
+{
+    internal class HealthBarGumpCustom : AnchorableGump
+    {
+
+        private const ushort LINE_RED = 0x0805;
+        private const ushort LINE_BLUE = 0x0806;
+        private const ushort LINE_POISONED = 0x0808;
+        private const ushort LINE_YELLOWHITS = 0x0809;
+        private const int MIN_WIDTH = 120;
+
+        private readonly GumpPicWithWidth[] _bars = new GumpPicWithWidth[3];
+        private GumpPic _hpLineRed, _manaLineRed, _stamLineRed;
+
+        private bool _canChangeName;
+        private bool _isDead;
+        private bool _oldWarMode, _normalHits, _poisoned, _yellowHits;
+        private bool _outOfRange;
+        private bool _targetBroke;
+
+        private string _name;
+        private int _oldHits, _oldStam, _oldMana;
+        private TextBox _textBox;
+
+        public RenderedText _renderedText;
+        public AlphaBlendControl _background;
+
+        public HealthBarGumpCustom(Entity entity) : this()
+        {
+            if (Engine.Profile.Current.CustomBarsToggled == false)
+            {
+                Engine.UI.GetGump<HealthBarGump>(entity.Serial);
+            }
+
+            Engine.UI.GetGump<HealthBarGump>(entity.Serial)?.Dispose();
+
+            if (entity == null)
+            {
+                Dispose();
+                return;
+            }
+
+            _name = entity.Name;
+            _isDead = entity.Serial.IsMobile && ((Mobile)entity).IsDead;
+            LocalSerial = entity.Serial;
+            Hue hue = entity is Mobile m ? Notoriety.GetHue(m.NotorietyFlag) : (Hue)0x0481;
+            _renderedText = RenderedText.Create(String.Empty, hue, 0xFF, true, FontStyle.BlackBorder, TEXT_ALIGN_TYPE.TS_CENTER, 100, 30, true);
+
+            BuildCustomGump();
+        }
+
+        public HealthBarGumpCustom(Serial mob) : this(World.Get(mob))
+        {
+        }
+
+        public HealthBarGumpCustom() : base(0, 0)
+        {
+            CanMove = true;
+            AnchorGroupName = "healthbar";
+        }
+
+        public override int GroupMatrixWidth
+        {
+            get => Width;
+            protected set { }
+        }
+
+        public override int GroupMatrixHeight
+        {
+            get => Height;
+            protected set { }
+        }
+
+        public void Update()
```

Comment:
can you rename this in ForceUpdate() ?
This is a standard behaviour that im using now.

Reply:
Sure thing, edited it to ForceUpdate()